### PR TITLE
Fix #562

### LINKED ISSUE #562: Locale overrider support
# Feature request

## Thesis

So, the idea is very simple - use context manager to temporarily override current locale for provider's instance, just like that:

```python
>>> from mimesis import Person
>>> from mimesis import locales

>>> person = Person(locales.EN)
>>> person.full_name()
'Ozie Melton'

>>> with locales.override(person, locales.RU):
...     person.full_name()

'Симона Богданова'

>>> person.full_name()
'Waldo Foster'
```

## Reasoning

Sometimes we need only some data from other provider and creating an instance each time is not really good, so it's better just override locale.

### PR DESCRIPTION
**Pull request by lk-geimfari**
# I have made things!

## Checklist

- [x] I have read [contributing guidelines](https://github.com/lk-geimfari/mimesis/blob/master/CONTRIBUTING.rst)
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [x] I have added my changes to the [CHANGELOG.rst](https://github.com/lk-geimfari/mimesis/blob/master/CHANGELOG.rst)

## Related issues

- Closes #562

### CODE REVIEW

File: mimesis/locales.py [hunk 1]
```diff
@@ -31,3 +33,35 @@
 TR = 'tr'
 UK = 'uk'
 ZH = 'zh'
+
+
+class override(object):  # noqa
```
Comment (sobolevn) [inline mimesis/locales.py]: Why not a generator?

File: mimesis/locales.py [hunk 2]
```diff
@@ -31,3 +33,35 @@
...
+    """Context manager which allows overriding current locale.
+
+     .. note:: This feature does not works with :class:`~mimesis.Generic()`.
+        Usually, you don't need data from other locales,
+        when you use :class:`~mimesis.Generic()`.
+    """
+
+    def __init__(self, provider, locale: str = EN) -> None:
```
Comment (sobolevn) [inline mimesis/locales.py]: `provider` can be also typed.

File: mimesis/locales.py (hunk 1 above)
Comment (sobolevn) [reply mimesis/locales.py]: Yeap

File: mimesis/locales.py [hunk 3]
```diff
@@ -31,3 +37,40 @@
 TR = 'tr'
 UK = 'uk'
 ZH = 'zh'
+
+_LocaleDependProvider = typing.Union[
```
Comment (lk-geimfari) [inline mimesis/locales.py]: Not sure about this, but this is a temporary solution until we close #565.

File: mimesis/locales.py [hunk 4]
```diff
@@ -35,33 +37,28 @@
 ZH = 'zh'
 
 
-class override(object):  # noqa
+@contextlib.contextmanager
+def override(provider, locale: str = EN):
```
Comment (sobolevn) [inline mimesis/locales.py]: `-> Generator[Provider, None, None]`

**Pull request merged**